### PR TITLE
Support using math lib in wasm

### PR DESCRIPTION
**Pull request by SantyWang**
Re: #

### Changelog

* Support using math lib in wasm, Ignore some interface when build wasm.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by SantyWang** [comment]
@cocos-robot run test cases

**Comment by SantyWang** [comment]
@cocos-robot run test cases

File: native/cocos/math/Vec4.cpp [hunk 1]
```diff
@@ -329,10 +331,12 @@ const Vec4 Vec4::UNIT_Y = Vec4(0.0F, 1.0F, 0.0F, 0.0F);
 const Vec4 Vec4::UNIT_Z = Vec4(0.0F, 0.0F, 1.0F, 0.0F);
 const Vec4 Vec4::UNIT_W = Vec4(0.0F, 0.0F, 0.0F, 1.0F);
 
+#if CC_PLATFORM != CC_PLATFORM_EMSCRIPTEN
 template <>
 ccstd::hash_t Hasher<Vec4>::operator()(const Vec4 &v) const {
     return ccstd::hash_range(reinterpret_cast<const uint64_t *>(&v.x),
                              reinterpret_cast<const uint64_t *>(&v.x + 4));
 }
+#endif
```
Comment (minggo) [inline native/cocos/math/Vec4.cpp:340]: wasm don't need this function?

Comment (SantyWang) [reply native/cocos/math/Vec4.cpp:340]: Only the basic functions of math are used, not the hash function.

Comment (minggo) [reply native/cocos/math/Vec4.cpp:340]: Got it. But new codes added in future may not compile in wasm mode. May be we should add a CI to make sure codes always work in wasm mode.

**Review by minggo — APPROVED** [review]

File: native/cocos/math/Vec4.cpp (hunk 1 above)
Comment (minggo) [reply native/cocos/math/Vec4.cpp:340]: And where to define `CC_PLATFORM_EMSCRIPTEN `?

Comment (SantyWang) [reply native/cocos/math/Vec4.cpp:340]: it's been defined in cmake.

Comment (SantyWang) [reply native/cocos/math/Vec4.cpp:340]: > Got it. But new codes added in future may not compile in wasm mode. May be we should add a CI to make sure codes always work in wasm mode.

We already have a wasm ci to check compiling result when building wasm.

**Pull request merged**